activesupport/lib/active_support/cache.rb

```diff
@@ -565,33 +563,13 @@ def merged_options(call_options)
         # Prefixes a key with the namespace. Namespace and key will be delimited
         # with a colon.
         def normalize_key(key, options)
-          key = expanded_key(key)
+          key = Cache.expand_cache_key(key)
           namespace = options[:namespace] if options
           prefix = namespace.is_a?(Proc) ? namespace.call : namespace
           key = "#{prefix}:#{key}" if prefix
           key
         end
 
-        # Expands key to be a consistent string value. Invokes +cache_key+ if
-        # object responds to +cache_key+. Otherwise, +to_param+ method will be
-        # called. If the key is a Hash, then keys will be sorted alphabetically.
-        def expanded_key(key)
-          return key.cache_key.to_s if key.respond_to?(:cache_key)
```

Comment:
Is there relevant history to why this method wasn't taught about `cache_key_with_version`?

Reply:
It goes back up to 7 years: there wasn't much change there and rails was not driven by united mind in that era.